tests/components/discovery/test_init.py

```diff
@@ -36,10 +37,12 @@ def netdisco_mock():
 
 async def mock_discovery(hass, discoveries, config=BASE_CONFIG):
     """Mock discoveries."""
-    result = await async_setup_component(hass, "discovery", config)
-    assert result
-
-    await hass.async_start()
+    with patch("homeassistant.components.zeroconf.async_get_instance"):
+        assert await async_setup_component(hass, "discovery", config)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)